streams/src/test/java/org/apache/kafka/streams/processor/internals/StreamTaskTest.java

```diff
@@ -2221,14 +2231,94 @@ public void shouldInitTaskTimeoutAndEventuallyThrow() {
     }
 
     @Test
-    public void shouldCLearTaskTimeout() {
+    public void shouldClearTaskTimeout() {
         task = createStatelessTask(createConfig());
 
         task.maybeInitTaskTimeoutOrThrow(0L, null);
         task.clearTaskTimeout();
         task.maybeInitTaskTimeoutOrThrow(Duration.ofMinutes(5).plus(Duration.ofMillis(1L)).toMillis(), null);
     }
 
+    @Test
+    public void shouldUpdateOffsetIfAllRecordsAreCorrupted() {
+        task = createStatelessTask(createConfig(AT_LEAST_ONCE,
+            "0",
+            LogAndContinueExceptionHandler.class.getName()));
+        task.initializeIfNeeded();
+        task.completeRestoration(noOpResetter -> { });
+
+        long offset = -1;
+        final List<ConsumerRecord<byte[], byte[]>> records = asList(
+            getCorruptedConsumerRecordWithOffsetAsTimestamp(++offset),
+            getCorruptedConsumerRecordWithOffsetAsTimestamp(++offset));
+        consumer.addRecord(records.get(0));
+        consumer.addRecord(records.get(1));
+        consumer.poll(Duration.ZERO);
```

Comment:
Why do we need to add records to the consumer and call `poll()` ? Why isn't `task.addRecords()` sufficient to setup the test?

Reply:
Consider `committableOffsetsAndMetadata`:

https://github.com/apache/kafka/blob/ed8b7875fed82feccec3d35aa700cce1ea5138f9/streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java#L438-L442

and

https://github.com/apache/kafka/blob/ed8b7875fed82feccec3d35aa700cce1ea5138f9/streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java#L453-L454

The objective of these test cases is to show that consumed offsets changed according to the logic. At the time of the assertions, there are no records in the `fifoQueue`, so `offset` is `null` in the line 438. Consequently, it falls back to checking the consumer's position. The consumer offset is updated on `poll`.

Reply:
Thanks!